pkg/providers/amifamily/windows.go

```diff
@@ -0,0 +1,82 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package amifamily
+
+import (
+	"fmt"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/karpenter/pkg/providers/amifamily/bootstrap"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/cloudprovider"
+
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+)
+
+type Windows struct {
+	DefaultFamily
+	*Options
+	template *v1alpha1.AWSNodeTemplate
+}
+
+// SSMAlias returns the AMI Alias to query SSM
+func (w Windows) SSMAlias(version string, _ *cloudprovider.InstanceType) string {
+	windowsVersion := getValueFromAnnotation(w.template.Annotations, v1alpha1.AnnoInstanceWindowsVersion, v1alpha1.Windows2019)
```

Comment:
Can we just specify a single Windows version as the default version and have anyone who wants to use a custom windows version use the `AMISelector` mechanism for picking AMIs?

Reply:
That works!! I remove the code of annotations.

Users can use AMISelector to choose the right Windows version (a document may be required).

Reply:
@jonathan-innis This might not be the ideal way. For Windows, the AMIs are updated monthly and therefore, if someone is using Karpenter then there would be additional overhead of figuring out the latest AMIs using the SSM parameter.

Also, each Windows version is it's own AMI family as per the customer usage. For example- Customers would have their Apps supported on Windows Server 2019 and they would hence run their workloads on Windows Server 2019 only. I don't think customers mix and match different versions of Windows in a single cluster.

As per the initial discussion with @bwagner5, we were considering one of the two approaches-
- We add `WindowsServer2019Full`, `WindowsServer2019Core`, ``WindowsServer2022Full`, and ``WindowsServer2022Core` as new AMI families. This is done by `eksctl` as well. Reference [here](https://eksctl.io/usage/schema/#nodeGroups-amiFamily). I am more inclined towards this approach.
- We make the users add annotations on pods about the target Windows version and edition. Then Karpenter would spin up the node as required for the given Windows version and edition.

What was done by @topikachu also works well in my opinion.

Reply:
> @jonathan-innis This might not be the ideal way. For Windows, the AMIs are updated monthly and therefore, if someone is using Karpenter then there would be additional overhead of figuring out the latest AMIs using the SSM parameter.
> 
> Also, each Windows version is it's own AMI family as per the customer usage. For example- Customers would have their Apps supported on Windows Server 2019 and they would hence run their workloads on Windows Server 2019 only. I don't think customers mix and match different versions of Windows in a single cluster.
> 
> As per the initial discussion with @bwagner5, we were considering one of the two approaches-
> 
> * We add `WindowsServer2019Full`, `WindowsServer2019Core`, `` WindowsServer2022Full`, and  ``WindowsServer2022Core`as new AMI families. This is done by`eksctl` as well. Reference [here](https://eksctl.io/usage/schema/#nodeGroups-amiFamily). I am more inclined towards this approach.

I like this. In fact, I create four Windows AMIFamiliy strings at the beginning but remove them and use annotation when filing the pull request.
I can add them back.
My implementation only needs to add 4 strings but uses a single Windows class to handle them all. 

> * We make the users add annotations on pods about the target Windows version and edition. Then Karpenter would spin up the node as required for the given Windows version and edition.
> 
> What was done by @topikachu also works well in my opinion.

Reply:
Done!
Add 4 new Windows Families.